### DISCUSSION TITLE
Scaling in the s5_energycurve.i file

### DISCUSSION THREAD

**Discussion by LiuPengPeter** [idaholab/moose#16379] [Q&A Modules: Phase field] [answered]
Hello,

I still couldn't understand the scaling factor and how it works on the residual.

In the s5_energycurve.i file, the scaling=1e+04 for variable c. And in the expression of material, kappa_c*1e-27, M/1e-27 and f_loc*1e-27.

Could anyone explain?

**Comment by jessecarterMOOSE** [comment]
1e-27 is for the _value_ of the property, sort of like a unit conversion. It's the same everywhere so it doesn't affect the solution. Probably @dschwen can tell you more about why that value is chosen. The variable scaling affects the residual. 

You might want to try the newer automatic scaling feature in moose. You can set `automatic_scaling=true` in your `Executioner` block and remove the `scaling` parameter for the variable.

**Reply by LiuPengPeter** [reply]
Than you!

**Comment by LiuPengPeter** [comment]
@dschwen , could you tell me more details about scaling.

1) Why and when does the scaling factor need to be added to variables and material parameters?

2)How the scaling factor influence the residual?
For example (maybe not right), there is a residual R1 whose variable is eta1 and coupled variable is eta2. M is the material parameter.
R1 = M*eta1^2 + eta2 + partial(f)/(eta1) +  partial(eta1)/(x).
After add scaling=1e9 to eta1, what will R1 become?
The R1 may be not right, I just want to use it as an example to know hoe the factor influence residual.

3)In the case of s5_energycurve.i file from tutorial of phase filed, the scaling=1e+04 for variable c. And in the expression of material, kappa_c1e-27, M/1e-27 and f_loc1e-27. Could you explain it?

Thank you a lot!

**Comment by LiuPengPeter** [comment, marked as answer]
I still couldn't understand the scaling factor...Could anyone help?

**Reply by aeslaughter** [reply]
@lindsayad Just noticed this sitting around. The only information that I can find on our auto scaling is here: https://mooseframework.inl.gov/newsletter/2019_07.html. Do we have more information somewhere else?

**Reply by lindsayad** [reply]
We have this documentation page: https://mooseframework.inl.gov/source/systems/NonlinearSystemBase.html. @LiuPengPeter can you see whether this helps you in your understanding?

**Reply by LiuPengPeter** [reply]
Than you all of guys!
